lib/katello/tasks/reset.rake

```diff
@@ -1,29 +1,57 @@
 namespace :katello do
   namespace :reset_backends do
-    service_stop = "sudo /sbin/service %s status > /dev/null && sudo /sbin/service %s stop"
-    service_start = "sudo /sbin/service %s start"
+    # Switch to foreman-maintain when Pulp3 services are added to it
+    service_stop = "sudo /usr/bin/systemctl status %s > /dev/null && sudo /usr/bin/systemctl stop %s"
+    service_start = "sudo /usr/bin/systemctl start %s"
 
-    task :pulp do
-      SERVICES = %w(httpd pulp_workers pulp_resource_manager pulp_celerybeat).freeze
-      system(service_stop.gsub("%s", "mongod"))
+    task :pulp_legacy do
+      SERVICES = %w(httpd pulp_workers pulp_resource_manager pulp_celerybeat squid qpidd pulp_streamer).freeze
+
+      puts "\e[33mStarting Pulp2 Reset\e[0m\n\n"
 
       SERVICES.each { |s| system(service_stop.gsub("%s", s)) }
-      system("sudo rm -rf /var/lib/mongodb/pulp_database*")
-      system(service_start.gsub("%s", "mongod"))
-      sleep(10)
-      fail "Cannot migrate pulp database" unless system("sudo -u apache /usr/bin/pulp-manage-db")
+      system(service_start.gsub("%s", 'rh-mongodb34-mongod'))
+      system("mongo pulp_database --eval 'db.dropDatabase();'")
+      fail "\e[31mCannot Migrate Pulp2 Database\e[0m\n\n" unless system('sudo -u apache /usr/bin/pulp-manage-db')
 
       SERVICES.each { |s| system(service_start.gsub("%s", s)) }
-      puts "Pulp database reset."
+      puts "\e[32mPulp2 Database Reset Complete\e[0m\n\n"
+    end
+
+    task :pulp do
+      # Pull active workers from Redis to see how many we need to start up
+      worker_count = `redis-cli --scan --pattern "rq:workers:reserved-resource-worker*" | wc -l`.chomp.to_i
+
+      services = %w(redis pulpcore-api pulpcore-resource-manager pulpcore-content)
+      while worker_count > 0
+        services.push("pulpcore-worker@#{worker_count}")
+        worker_count -= worker_count
+      end
+
+      puts "\e[33mStarting Pulp3 Reset\e[0m\n\n"
+
+      services.each { |s| system(service_stop.gsub("%s", s)) }
+      system("sudo runuser - postgres -c 'dropdb pulp -p 7878'")
+      system("sudo runuser - postgres -c 'createdb pulp -p 7878'")
+      # rubocop:disable Style/BlockDelimiters
+      Dir.chdir('/usr/local/lib/pulp/bin') {
+        fail "\e[31mCannot migrate Pulp3 database\e[0m\n\n" unless system("sudo -u pulp PULP_SETTINGS='/etc/pulp/settings.py' DJANGO_SETTINGS_MODULE='pulpcore.app.settings' ./django-admin migrate --no-input")
+        puts "\e[33mRecreating Admin User\e[0m\n\n"
+        system("sudo -u pulp PULP_SETTINGS='/etc/pulp/settings.py' DJANGO_SETTINGS_MODULE='pulpcore.app.settings' ./django-admin reset-admin-password --password password")
+      }
+      # rubocop:enable Style/BlockDelimiters
```

Comment:
is there a reason you didn't use block delimiters here? like:
```
Dir.chdir(path) do 
  Code
  Code 
end
```

Reply:
I can write it like that, didnt know it would work that way without the braces, all the examples on rubocop I saw were .each and .item. I will fix that too

Reply:
yep, all blocks can be written either as:
```
method {   
}
```

or 
```
method do 
end
```

These two ways of doing it are exactly the same.

we don't allow {} for multi-line blocks as a style rule, meaning its only allowed when its written in one line:
```
method {    }
```